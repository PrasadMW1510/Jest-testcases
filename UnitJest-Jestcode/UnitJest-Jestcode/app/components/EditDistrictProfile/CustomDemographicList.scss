.manage-admin-accounts__header-text {
  margin-top: 3px;
  margin-bottom: 3px;
}

.manage-admin-accounts__text-direction {
  font-size: 12px;
}

.manage-admin-accounts__link-container {
  display: flex;
  font-size: 12px;
  width: 450px;
  margin-bottom: 7px;
  margin-top: 7px;
  justify-content: space-between;
}

.manage-admin-accounts__link-section {
  display: flex;
  flex-direction: column;
}

.custom-demographic-list__table-link {
  cursor: pointer;
  color: #00f;
  margin-bottom: 7px;
  text-decoration: underline;
  text-decoration: underline;
}

// Styles that extend the styles provided by ReactTable/SAMTable
$thead-color: #311c00;
$thead-background-color: #ffd400;
$thead-clicked-color: #ffa200;
$odd-table-row-background-color: #ffef9e;
$even-table-row-background-color: #ffef9e;

/* stylelint-disable-next-line selector-class-pattern */
.custom-demographics-table > .ReactTable {
  height: 200px;
  font-size: 12px;
  margin-bottom: 30px;
  border-top: initial;

  // these styles are added by javascript after a table header was clicked
  .-sort-asc,
  .-sort-desc {
    box-shadow: inset 1px 1px 1px 1px rgba(0, 0, 0, 0.1);
    background-color: $thead-clicked-color;
  }

  .rt-thead {
    background-color: $thead-background-color;
    border-bottom: initial;
    color: $thead-color;
    font-weight: 700;
    height: 24px;
    max-height: 24px;
  }

  .rt-th {
    display: flex;
    flex: 1;
    font-size: 11px;
    max-height: 24px;
    overflow: hidden;
    padding-left: 10px;
    padding-top: 6px;
    text-overflow: ellipsis;
    border-style: none;
    transition: 0.3s ease;
    transition-property: width, min-width, padding, opacity;
  }

  .rt-tbody {
    border-right: 1px solid #000;
    background-color: #ffef9e;
  }

  .rt-tbody:last-child {
    border: initial;
  }

  .rt-td {
    flex: 1;
    padding: 3px 10px;
    text-overflow: ellipsis;
    overflow: hidden;
    transition: 0.3s ease;
    border-right: 0 solid #000;
    background-color: #ffef9e;
    transition-property: width, min-width, padding, opacity;
  }

  .rt-td:last-child {
    border: initial;
    color: #00f;
    text-decoration: underline;
  }

  .rt-tr {
    border-bottom: 1px solid #ffb200;
    max-height: 30px;
    text-overflow: ellipsis;
  }

  .rt-tr.-odd {
    background: $odd-table-row-background-color;
  }

  .rt-tr-group {
    max-height: 25px;
    background-color: #ffef9e;
  }

  .rt-tr.-even {
    background: $even-table-row-background-color;
  }

  // styles for custom elements within SAMTable
  .custom-demographics-table__body {
    border-bottom: 1px solid #000;
    border-right: 1px solid #000;
    display: block;
    height: 185px;
    overflow-y: auto;
    width: 772px;
  }

  .custom-demographics-table__no-data {
    background-color: #ffef9e;
    display: inherit;
    height: 188px;
  }

  .custom-demographics-table__no-data-text {
    color: #f00;
    font-size: 12px;
    margin-left: 12px;
    margin-top: 70px;
    text-align: center;
  }

  .custom-demographics-table__cell-enabled {
    text-align: center;
  }

  .custom-demographics-table__header-enabled {
    background-color: #ffd400;
  }
}
