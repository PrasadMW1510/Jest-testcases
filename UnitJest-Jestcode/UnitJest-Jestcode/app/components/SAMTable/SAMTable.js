/**
 *
 * SAMTable
 * This is the generic table that other tables extend over.
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { sortData } from 'utils/utilities';

// Import React Table
import ReactTable, { ReactTableDefaults } from 'react-table';
import checkboxHOC from 'react-table/lib/hoc/selectTable';
import './SAMTable.scss';

const CheckboxTable = checkboxHOC(ReactTable);

class SAMTable extends React.Component {
  isSelected = id => this.props.checkedIds.includes(id);

  toggleAll = () => {
    const selection = [];
    if (!this.props.selectAll) {
      // we need to get at the internals of ReactTable
      const wrappedInstance = this.checkboxTable.getWrappedInstance();
      // the 'sortedData' property contains the currently accessible records based on the filter and sort
      const currentRecords = wrappedInstance.getResolvedState().sortedData;
      currentRecords.forEach(item => {
        // eslint-disable-next-line no-underscore-dangle
        selection.push(item._original._id);
      });
    }

    this.props.toggleAllCheckboxes(!this.props.selectAll, selection);
  };

  toggleSelection = key => {
    const selection = [...this.props.checkedIds];
    const keyIndex = selection.indexOf(key);
    let checked = true;

    if (keyIndex >= 0) {
      checked = false;
    }
    this.props.handleRowCheckboxOnChange(checked, key);
  };

  defaultSortMethod = (a, b) => sortData(a, b);

  render() {
    const {
      columns,
      className,
      hasCheckboxes,
      data,
      renderEmptyTable,
      selectAll,
      pageSize,
      sortable,
    } = this.props;
    // If there is no data, renderEmptyTable element is rendered
    // as a body of the table
    let body = renderEmptyTable;

    if (data && data.length > 0) {
      body = ReactTableDefaults.TbodyComponent;
    }

    return (
      <div className={className}>
        {!hasCheckboxes && (
          <ReactTable
            data={data}
            columns={columns}
            resizable={false}
            pageSize={pageSize}
            showPagination={false}
            sortable={sortable}
            TbodyComponent={body}
            defaultSortMethod={this.defaultSortMethod}
            {...this.props}
          />
        )}
        {hasCheckboxes && (
          <CheckboxTable
            ref={node => (this.checkboxTable = node)}
            data={
              data // eslint-disable-line no-return-assign
            }
            columns={columns}
            selectType="checkbox"
            resizable={false}
            toggleSelection={this.toggleSelection}
            selectAll={selectAll}
            isSelected={this.isSelected}
            toggleAll={this.toggleAll}
            pageSize={pageSize}
            showPagination={false}
            defaultSortMethod={this.defaultSortMethod}
            TbodyComponent={body}
            {...this.props}
          />
        )}
      </div>
    );
  }
}

SAMTable.defaultProps = {
  checkedIds: [],
  data: [],
  selectAll: false,
  pageSize: 0,
  sortable: true,
};

SAMTable.propTypes = {
  checkedIds: PropTypes.array,
  className: PropTypes.string,
  columns: PropTypes.array,
  data: PropTypes.array,
  hasCheckboxes: PropTypes.bool,
  renderEmptyTable: PropTypes.func,
  handleRowCheckboxOnChange: PropTypes.func,
  toggleAllCheckboxes: PropTypes.func,
  selectAll: PropTypes.bool,
  sortable: PropTypes.bool,
  pageSize: PropTypes.number,
};

export default SAMTable;
