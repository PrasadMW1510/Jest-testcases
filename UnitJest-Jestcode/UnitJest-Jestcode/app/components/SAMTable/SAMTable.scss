/*
** Styles inherited from
** https://github.com/react-tools/react-table
** React-Table original styles do not follow
** selector-class-pattern
*/

/* stylelint-disable selector-class-pattern */

.ReactTable {
  position: relative;
  display: flex;
  flex-direction: column;
  border: 1px solid #000;
}

.ReactTable * {
  box-sizing: border-box;
}

.ReactTable .rt-table {
  flex: auto 1;
  display: flex;
  flex-direction: column;
  align-items: stretch;
  width: 100%;
  border-collapse: collapse;
  overflow: auto;
  height: 200px;
}

.ReactTable .rt-thead {
  flex: 1 0 auto;
  display: flex;
  flex-direction: column;
  user-select: none;
}

.ReactTable .rt-thead.-filters input,
.ReactTable .rt-thead.-filters select {
  border: 1px solid rgba(0, 0, 0, 0.1);
  background: #fff;
  padding: 5px 7px;
  font-size: inherit;
  border-radius: 3px;
  font-weight: normal;
  outline: none;
}

.ReactTable .rt-thead.-filters .rt-th {
  border-right: 1px solid #000;
}

.ReactTable .rt-thead.-header {
  box-shadow: 0 2px 15px 0 rgba(0, 0, 0, 0.15);
}

.ReactTable .rt-thead .rt-tr {
  text-align: center;
}

.ReactTable .rt-thead .rt-th,
.ReactTable .rt-thead .rt-td {
  padding: 5px;
  line-height: normal;
  position: relative;
  border-right: 1px solid #000;
  transition: box-shadow 0.3s cubic-bezier(0.175, 0.885, 0.32, 1.275);
  box-shadow: inset 0 0 0 0 transparent;
}

.ReactTable .rt-thead .rt-th.-cursor-pointer,
.ReactTable .rt-thead .rt-td.-cursor-pointer {
  cursor: pointer;
}

.ReactTable .rt-thead .rt-th:last-child,
.ReactTable .rt-thead .rt-td:last-child {
  border-right: 0;
}

.ReactTable .rt-thead .rt-header-pivot {
  border-right-color: rgb(129, 63, 63);
}

.rt-thead .rt-header-pivot,
.rt-thead .rt-header-pivot {
  ::after,
  ::before {
    left: 100%;
    top: 50%;
    border: solid transparent;
    content: ' ';
    height: 0;
    width: 0;
    position: absolute;
    pointer-events: none;
  }

}

.rt-thead .rt-header-pivot {
  ::after {
    border-color: rgba(255, 255, 255, 0);
    border-left-color: #fff;
    border-width: 8px;
    margin-top: -8px;
  }
}

.rt-thead .rt-header-pivot {
  ::before {
    border-color: rgba(102, 102, 102, 0);
    border-left-color: #f7f7f7;
    border-width: 10px;
    margin-top: -10px;
  }
}

.ReactTable .rt-tbody {
  flex: 99999 1 auto;
  display: flex;
  flex-direction: column;
  overflow: auto;
}

.ReactTable .rt-tbody .rt-tr-group:last-child {
  border-bottom: 0;
}

.ReactTable .rt-tbody .rt-td:last-child {
  border-right: 0;
}

.ReactTable .rt-tbody .rt-expandable {
  cursor: pointer;
}

.ReactTable .rt-tr-group {
  flex: 1 0 auto;
  display: flex;
  flex-direction: column;
  align-items: stretch;
}

.ReactTable .rt-tr {
  flex: 1 0 auto;
  display: inline-flex;
}

.ReactTable .rt-noData {
  display: none;
}

.ReactTable .rt-expander.-open {
  ::after {
    transform: translate(-50%, -50%) rotate(0);
  }
}

.ReactTable .rt-tfoot {
  flex: 1 0 auto;
  display: flex;
  flex-direction: column;
  box-shadow: 0 0 15px 0 rgba(0, 0, 0, 0.15);
}

.ReactTable .rt-tfoot .rt-td {
  border-right: 1px solid #000;
}

.ReactTable .rt-tfoot .rt-td:last-child {
  border-right: 0;
}

.ReactTable .-loading {
  display: none;
}

// Use this definition to truncate data with an ellipsis
// inside a react-table cell
.rt-td__truncated-block {
  flex: 1;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

// Tooltips used inside table cells

// NOTE: Use overrides in separate CSS definitions to change
// the tooltip styling for specific components.

.rt-td__tooltip {
  display: block;
  overflow-wrap: break-word;
  position: relative;
  white-space: normal;
}

.rt-td__tooltip--right:hover::after {
  background: #ffdd69;
  border-radius: 5px;
  bottom: -5px;
  color: #000;
  content: attr(data-tip);
  display: flex;
  left: 20%;
  padding: 5px 15px;
  position: absolute;
  z-index: 98;
}

.rt-td__tooltip--left:hover::after {
  background: #ffdd69;
  border-radius: 5px;
  color: #000;
  content: attr(data-tip);
  left: -200px;
  padding: 5px 15px;
  position: absolute;
  z-index: 98;
  width: 220px;
}

/* stylelint-enable selector-class-pattern
*/
