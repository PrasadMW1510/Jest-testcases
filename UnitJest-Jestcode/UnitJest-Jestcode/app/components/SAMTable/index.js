/**
 *
 * SAMTable
 *
 */

export { default } from './SAMTable';
