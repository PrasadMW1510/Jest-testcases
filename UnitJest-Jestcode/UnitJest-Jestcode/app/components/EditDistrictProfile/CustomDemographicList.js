/**
 *
 * CustomDemographicList
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import * as ModalConstants from 'containers/ModalController/constants';
import SAMTable from 'components/SAMTable';
import './CustomDemographicList.scss';

class CustomDemographicList extends React.Component {
  constructor(props) {
    super(props);
    this.table = null;
  }

  handleEditDemographic = demographicToEdit => {
    const selectedIndex = this.props.demographics.findIndex(
      demographic => demographic.name === demographicToEdit
    );

    this.props.showModal(ModalConstants.ADD_DEMOGRAPHIC_MODAL, {
      editMode: true,
      demographicToEdit,
      demographicList: this.props.demographics,
      selectedIndex,
    });
  };

  handleDeleteDemographic = demographicToDelete => {
    const selectedIndex = this.props.demographics.findIndex(
      demographic => demographic.name === demographicToDelete
    );

    this.props.showModal(ModalConstants.ADD_DEMOGRAPHIC_MODAL, {
      deleteMode: true,
      demographicToDelete,
      demographicList: this.props.demographics,
      selectedIndex,
    });
  };
  renderEmptyTable = () => (
    <div className="custom-demographics-table__no-data">
      <div className="custom-demographics-table__no-data-text" />
    </div>
  );

  renderTableData = () =>
    this.props.demographics.map(rowData => ({
      _id: rowData.name,
      ...rowData,
    }));

  render() {
    const columns = [
      {
        Header: 'Custom',
        id: 'name',
        width: 180,
        accessor: 'name',
      },
      {
        Header: '',
        width: 100,
        accessor: 'edit',
        Cell: row => (
          <span
            role="link"
            className="custom-demographic-list__table-link"
            tabIndex={0}
            onClick={() => this.handleEditDemographic(row.original.name)}
          >
            edit
          </span>
        ),
      },
      {
        Header: '',
        accessor: 'delete',
        width: 100,
        Cell: row => (
          <span
            role="link"
            className="custom-demographic-list__table-link"
            tabIndex={0}
            onClick={() => this.handleDeleteDemographic(row.original.name)}
          >
            delete
          </span>
        ),
      },
    ];
    return (
      <SAMTable
        ref={node => (this.table = node)}
        className="custom-demographics-table"
        columns={columns}
        data={this.renderTableData()}
        renderEmptyTable={this.renderEmptyTable}
        pageSize={this.props.demographics.length}
        sortable={false}
      />
    );
  }
}

CustomDemographicList.defaultProps = {
  demographics: [],
};

CustomDemographicList.propTypes = {
  demographics: PropTypes.array,
  showModal: PropTypes.array,
};

export default CustomDemographicList;
