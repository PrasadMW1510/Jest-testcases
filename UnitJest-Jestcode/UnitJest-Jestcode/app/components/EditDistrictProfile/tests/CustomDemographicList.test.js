import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SAMTable from 'components/SAMTable';
import CustomDemographicList from '../CustomDemographicList';

describe('<CustomDemographicList />', () => {
  let showModalSpy;
  beforeEach(() => {
    showModalSpy = jest.fn();
  });

  it('Should render correctly with no data', () => {
    const wrapper = mount(<CustomDemographicList demographics={[]} showModal={showModalSpy} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should populate the Custom Demographics Table with the right columns', () => {
    const mockData = {
      name: 'Test Demographics',
    };
    const mockRowData = {
      original: mockData,
    };
    const wrapper = shallow(
      <CustomDemographicList demographics={[mockData]} showModal={showModalSpy} />
    );
    const table = wrapper.find(SAMTable);
    const { columns } = table.props();

    const editColumn = columns.find(row => row.accessor === 'edit');
    const editColumnWrapper = shallow(editColumn.Cell(mockRowData));

    const deleteColumn = columns.find(row => row.accessor === 'delete');
    const deleteColumnWrapper = shallow(deleteColumn.Cell(mockRowData));

    expect(shallowToJson(editColumnWrapper)).toMatchSnapshot();
    expect(shallowToJson(deleteColumnWrapper)).toMatchSnapshot();
  });
});
