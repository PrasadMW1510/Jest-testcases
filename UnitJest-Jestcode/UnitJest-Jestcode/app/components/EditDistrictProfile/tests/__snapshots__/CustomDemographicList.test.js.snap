// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomDemographicList /> Should render correctly with no data 1`] = `
<CustomDemographicList
  demographics={Array []}
  showModal={[Function]}
>
  <SAMTable
    checkedIds={Array []}
    className="custom-demographics-table"
    columns={
      Array [
        Object {
          "Header": "Custom",
          "accessor": "name",
          "id": "name",
          "width": 180,
        },
        Object {
          "Cell": [Function],
          "Header": "",
          "accessor": "edit",
          "width": 100,
        },
        Object {
          "Cell": [Function],
          "Header": "",
          "accessor": "delete",
          "width": 100,
        },
      ]
    }
    data={Array []}
    pageSize={0}
    renderEmptyTable={[Function]}
    selectAll={false}
    sortable={false}
  >
    <div
      className="custom-demographics-table"
    >
      <ReactTable
        AggregatedComponent={[Function]}
        ExpanderComponent={[Function]}
        FilterComponent={[Function]}
        LoadingComponent={[Function]}
        NoDataComponent={[Function]}
        PadRowComponent={[Function]}
        PaginationComponent={[Function]}
        PivotValueComponent={[Function]}
        ResizerComponent={[Function]}
        TableComponent={[Function]}
        TbodyComponent={[Function]}
        TdComponent={[Function]}
        TfootComponent={[Function]}
        ThComponent={[Function]}
        TheadComponent={[Function]}
        TrComponent={[Function]}
        TrGroupComponent={[Function]}
        aggregatedKey="_aggregated"
        checkedIds={Array []}
        className="custom-demographics-table"
        collapseOnDataChange={true}
        collapseOnPageChange={true}
        collapseOnSortingChange={true}
        column={
          Object {
            "Aggregated": undefined,
            "Cell": undefined,
            "Expander": undefined,
            "Filter": undefined,
            "Footer": undefined,
            "Header": undefined,
            "Pivot": undefined,
            "PivotValue": undefined,
            "aggregate": undefined,
            "className": "",
            "filterAll": false,
            "filterMethod": undefined,
            "filterable": undefined,
            "footerClassName": "",
            "footerStyle": Object {},
            "getFooterProps": [Function],
            "getHeaderProps": [Function],
            "getProps": [Function],
            "headerClassName": "",
            "headerStyle": Object {},
            "minWidth": 100,
            "resizable": undefined,
            "show": true,
            "sortMethod": undefined,
            "sortable": undefined,
            "style": Object {},
          }
        }
        columns={
          Array [
            Object {
              "Header": "Custom",
              "accessor": "name",
              "id": "name",
              "width": 180,
            },
            Object {
              "Cell": [Function],
              "Header": "",
              "accessor": "edit",
              "width": 100,
            },
            Object {
              "Cell": [Function],
              "Header": "",
              "accessor": "delete",
              "width": 100,
            },
          ]
        }
        data={Array []}
        defaultExpanded={Object {}}
        defaultFilterMethod={[Function]}
        defaultFiltered={Array []}
        defaultPageSize={20}
        defaultResized={Array []}
        defaultSortDesc={false}
        defaultSortMethod={[Function]}
        defaultSorted={Array []}
        expanderDefaults={
          Object {
            "filterable": false,
            "resizable": false,
            "sortable": false,
            "width": 35,
          }
        }
        filterable={false}
        freezeWhenExpanded={false}
        getLoadingProps={[Function]}
        getNoDataProps={[Function]}
        getPaginationProps={[Function]}
        getProps={[Function]}
        getResizerProps={[Function]}
        getTableProps={[Function]}
        getTbodyProps={[Function]}
        getTdProps={[Function]}
        getTfootProps={[Function]}
        getTfootTdProps={[Function]}
        getTfootTrProps={[Function]}
        getTheadFilterProps={[Function]}
        getTheadFilterThProps={[Function]}
        getTheadFilterTrProps={[Function]}
        getTheadGroupProps={[Function]}
        getTheadGroupThProps={[Function]}
        getTheadGroupTrProps={[Function]}
        getTheadProps={[Function]}
        getTheadThProps={[Function]}
        getTheadTrProps={[Function]}
        getTrGroupProps={[Function]}
        getTrProps={[Function]}
        groupedByPivotKey="_groupedByPivot"
        indexKey="_index"
        loading={false}
        loadingText="Loading..."
        multiSort={true}
        nestingLevelKey="_nestingLevel"
        nextText="Next"
        noDataText="No rows found"
        ofText="of"
        onFetchData={[Function]}
        originalKey="_original"
        pageSize={0}
        pageSizeOptions={
          Array [
            5,
            10,
            20,
            25,
            50,
            100,
          ]
        }
        pageText="Page"
        pivotDefaults={Object {}}
        pivotIDKey="_pivotID"
        pivotValKey="_pivotVal"
        previousText="Previous"
        renderEmptyTable={[Function]}
        resizable={false}
        rowsText="rows"
        selectAll={false}
        showPageJump={true}
        showPageSizeOptions={true}
        showPagination={false}
        showPaginationBottom={true}
        showPaginationTop={false}
        sortable={false}
        style={Object {}}
        subRowsKey="_subRows"
      >
        <div
          className="ReactTable custom-demographics-table"
          style={Object {}}
        >
          <Table
            className=""
          >
            <div
              className="rt-table"
            >
              <Thead
                className="-header"
                style={
                  Object {
                    "minWidth": "380px",
                  }
                }
              >
                <div
                  className="rt-thead -header"
                  style={
                    Object {
                      "minWidth": "380px",
                    }
                  }
                >
                  <Tr>
                    <div
                      className="rt-tr"
                    >
                      <ThComponent
                        className=""
                        key="0-name"
                        style={
                          Object {
                            "flex": "180 0 auto",
                            "maxWidth": "180px",
                            "width": "180px",
                          }
                        }
                        toggleSort={[Function]}
                      >
                        <div
                          className="rt-th"
                          onClick={[Function]}
                          role="heading"
                          style={
                            Object {
                              "flex": "180 0 auto",
                              "maxWidth": "180px",
                              "width": "180px",
                            }
                          }
                        >
                          <div
                            className=""
                          >
                            Custom
                          </div>
                        </div>
                      </ThComponent>
                      <ThComponent
                        className=""
                        key="1-edit"
                        style={
                          Object {
                            "flex": "100 0 auto",
                            "maxWidth": "100px",
                            "width": "100px",
                          }
                        }
                        toggleSort={[Function]}
                      >
                        <div
                          className="rt-th"
                          onClick={[Function]}
                          role="heading"
                          style={
                            Object {
                              "flex": "100 0 auto",
                              "maxWidth": "100px",
                              "width": "100px",
                            }
                          }
                        >
                          <div
                            className=""
                          />
                        </div>
                      </ThComponent>
                      <ThComponent
                        className=""
                        key="2-delete"
                        style={
                          Object {
                            "flex": "100 0 auto",
                            "maxWidth": "100px",
                            "width": "100px",
                          }
                        }
                        toggleSort={[Function]}
                      >
                        <div
                          className="rt-th"
                          onClick={[Function]}
                          role="heading"
                          style={
                            Object {
                              "flex": "100 0 auto",
                              "maxWidth": "100px",
                              "width": "100px",
                            }
                          }
                        >
                          <div
                            className=""
                          />
                        </div>
                      </ThComponent>
                    </div>
                  </Tr>
                </div>
              </Thead>
              <Component
                className=""
                style={
                  Object {
                    "minWidth": "380px",
                  }
                }
              >
                <div
                  className="custom-demographics-table__no-data"
                >
                  <div
                    className="custom-demographics-table__no-data-text"
                  />
                </div>
              </Component>
            </div>
          </Table>
          <NoData>
            <div
              className="rt-noData"
            >
              No rows found
            </div>
          </NoData>
          <LoadingComponent
            loading={false}
            loadingText="Loading..."
          >
            <div
              className="-loading"
            >
              <div
                className="-loading-inner"
              >
                Loading...
              </div>
            </div>
          </LoadingComponent>
        </div>
      </ReactTable>
    </div>
  </SAMTable>
</CustomDemographicList>
`;

exports[`<CustomDemographicList /> should populate the Custom Demographics Table with the right columns 1`] = `
<span
  className="custom-demographic-list__table-link"
  onClick={[Function]}
  role="link"
  tabIndex={0}
>
  edit
</span>
`;

exports[`<CustomDemographicList /> should populate the Custom Demographics Table with the right columns 2`] = `
<span
  className="custom-demographic-list__table-link"
  onClick={[Function]}
  role="link"
  tabIndex={0}
>
  delete
</span>
`;
